utils/preparation_data.py

```diff
@@ -0,0 +1,170 @@
+    CityScapesClass("sky", 23, (70, 130, 180)),
+    CityScapesClass("person", 24, (220, 20, 60)),
+    CityScapesClass("rider", 25, (225, 0, 0)),
+    CityScapesClass("car", 26, (0, 0, 142)),
+    CityScapesClass("truck", 27, (0, 0, 70)),
+    CityScapesClass("bus", 28, (0, 60, 100)),
+    CityScapesClass("caravan", 29, (0, 0, 90)),
+    CityScapesClass("trailer", 30, (0, 0, 110)),
+    CityScapesClass("train", 31, (0, 80, 100)),
+    CityScapesClass("motorcycle", 32, (0, 0, 230)),
+    CityScapesClass("bicycle", 33, (119, 11, 32)),
+    CityScapesClass("license plate", -1, (0, 0, 142)),
+]
+
+
+def open_zip_files(zip_dataset_paths: list[str], output_path: str) -> None:
+    """
+    Extract contents of multiple zip files to a directory.
+    """
+    for zip_dataset_path in zip_dataset_paths:
+        with zipfile.ZipFile(zip_dataset_path, "r") as zip_ref:
+            zip_ref.extractall(output_path + "\\temp\\")
+
+
+def copy_images(output_path: str) -> None:
+    """
+    Copy images and masks from input_zip_file to input_path.
+    """
+    TEMP_PATH = output_path + "\\temp\\"
+    IMG_SUFFIX = "_leftImg8bit.png"
+    MASK_SUFFIX = "_gtFine_labelIds.png"
+
+    SUFFIX_REPLACEMENT = {
+        IMG_SUFFIX: "_input.png",
+        MASK_SUFFIX: "_mask.png",
+    }
+
+    for root, _, files in os.walk(TEMP_PATH):
+        # Skip test folder because it does not have correct mask
+        if "test" not in root:
+            for file in files:
+                # Choose images
+                # if img.endswith(IMG_SUFFIX) or img.endswith(MASK_SUFFIX):
+                # replace the suffix with the new suffix
+                for suffix, replacement in SUFFIX_REPLACEMENT.items():
+                    if file.endswith(suffix):
+                        new_filename = file.replace(suffix, replacement)
+                        original_path = os.path.join(root, file)
+                        destination_path = os.path.join(output_path, new_filename)
+                        shutil.move(original_path, destination_path)
+                        break
+
+    # Remove the empty directories
+    shutil.rmtree(TEMP_PATH)
+
+
+def create_masks_by_class(output_path: str, class_to_keep: list[str] = []):
+
+    if class_to_keep:
+        # Get the labels of the classes to keep
+        labels = [cls.id for cls in CityScapesClasses if cls.name in class_to_keep]
+    else:
+        # Keep all classes
+        labels = [cls.id for cls in CityScapesClasses]
+
+    for image_file in os.listdir(output_path):
+        if image_file.endswith("_mask.png"):
+            image_path = os.path.join(output_path, image_file)
+            # Read the image using cv2.imread
+            image = cv2.imread(image_path, cv2.IMREAD_GRAYSCALE)
+            mask = np.isin(image, labels)
+            mask_value = 0
+            cleaned_mask = np.where(mask, image, mask_value)
+            mask_file_path = os.path.join(output_path, image_file)
```

Comment:
If `class_to_keep` is empty - why are we still preparing lables and using the `np.isin(image, labels)` ?
We keep all classes/ids, so there is no need for that in that case.